lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -128,10 +152,230 @@ hashSeed :: ScrubbedBytes -> ScrubbedBytes
+|                                                                                |
++--------------------------------------------------------------------------------+
+                                    |
+                                    |
+                                    v
++--------------------------------------------------------------------------------+
+|     Address Root    |     Address Attributes    |           AddrType           |
+|                     |                           |                              |
+|   Hash (224 bits)   |  Der. Path* + Stake + NM  |  PubKey | (Script) | Redeem  |
+|                     |    (open for extension)   |     (open for extension)     |
++--------------------------------------------------------------------------------+
+             |                 |
+             |                 |     +----------------------------------+
+             v                 |     |        Derivation Path           |
++---------------------------+  |---->|                                  |
+|  SHA3-256 >> Blake2b_224  |  |     | ChaChaPoly* AccountIx/AddressIx  |
+|                           |  |     +----------------------------------+
+|  -AddrType                |  |
+|  -ASD* (~AddrType+PubKey) |  |     +----------------------------------+
+|  -Address Attributes      |  |     |       Stake Distribution         |
++---------------------------+  |     |                                  |
+                               |---->|  BootstrapEra | (Single | Multi) |
+                               |     +----------------------------------+
+                               |
+                               |     +----------------------------------+
+                               |     |          Network Magic           |
+                               |---->|                                  |
+                                     | Addr Discr: MainNet vs TestNet   |
+                                     +----------------------------------+
+
+-------------------------------------------------------------------------------}
+
+{-
+-- The full address generation function, for reference.
+rndToAddress
+    :: Key 'AddressK XPub
+    -> Key 'RootK XPub
+    -> Index 'Hardened 'AccountK
+    -> Index 'Soft 'AddressK
+    -> Address
+rndToAddress (Key addrXPub) rootKey accIx addrIx =
+    Address $ CBOR.toStrictByteString $
+    encodeAddress addrXPub $
+    encodeAddressAttributes rootKey accIx addrIx
+
+encodeAddressAttributes
+    :: Key 'RootK XPub
+    -> Index 'Hardened 'AccountK
+    -> Index 'Soft 'AddressK
+    -> CBOR.Encoding
+encodeAddressAttributes rootKey accIx addrIx = mempty
+    <> CBOR.encodeMapLen 1
+    <> CBOR.encodeWord8 1
+    <> encodeDerivationPath rootKey accIx addrIx
+-}
+
+-- | Encode the derivation path (account index and address index) part of a
+-- random HD address.
+--
+-- This is the opposite of 'decodeDerivationPath'.
+--
+-- NOTE: The caller must ensure adequate key length.
```

Comment:
Would be nice to indicate what is an "adequate key length" :stuck_out_tongue:  
Also, if that's something that could make the program misbehave, we should consider adding some invariant check in the function body, do you agree?

Reply:
I have updated the comment to say public key length must be 32 bytes.
The encryption is going to fail if the key length is wrong, which results in an `error` call.
Adding an `invariant` check will also result in an `error` call.
It's the same.